### PR TITLE
Add info.urbanterror.UrbanTerror

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://www.urbanterror.info/forums/topic/36240-flathub/
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/FrozenSand/ioq3-for-UrbanTerror-4/pull/87

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build info.urbanterror.UrbanTerror

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing

File: info.urbanterror.UrbanTerror.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+app-id: info.urbanterror.UrbanTerror
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+branch: stable
```
Comment (hfiguiere) [inline info.urbanterror.UrbanTerror.yaml:5]:
```suggestion
```

File: info.urbanterror.UrbanTerror.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+rename-icon: urbanterror
+rename-desktop-file: urbanterror.desktop
+rename-appdata-file: urbanterror.appdata.xml
+finish-args:
+  - --socket=pulseaudio
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --extension=org.freedesktop.Platform.GL=directory=lib/GL
```
Comment (hfiguiere) [inline info.urbanterror.UrbanTerror.yaml:16]: this is not how you specify an extension.

I don't think you need it anyway.

File: urbanterror.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component type="desktop">
+  <id>urbanterror.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+ AND LicenseRef-proprietary=https://www.urbanterror.info/support/202-quake-3-sdk/</project_license>
```
Comment (hfiguiere) [inline urbanterror.appdata.xml:5]: it is GPL or is it proprietary?

Note: game data isn't considered here.

File: info.urbanterror.UrbanTerror.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        commit: cabd3370b32fb46dba33a8eda79162448923d896
+
+  - name: urbanterror
+    buildsystem: simple
+    build-commands:
+      - rm -rf Quake3-UrT*
+      - mkdir -p ${FLATPAK_DEST}/share/urbanterror
+      - mv q3ut4 ${FLATPAK_DEST}/share/urbanterror/q3ut4
+      - install -Dm0644 urbanterror.desktop ${FLATPAK_DEST}/share/applications/urbanterror.desktop
+      - install -Dm0644 urbanterror.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/urbanterror.png
```
Comment (hfiguiere) [inline info.urbanterror.UrbanTerror.yaml:49]: you never install the appdata file.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Plase make sure to submit the appdata.xml, .desktop and icon upstream for inclusion.

The lack of .desktop upstream IS a bug.

File: flathub.json (hunk 1 above)
Comment (Mailaender) [reply flathub.json:2]: yes, but https://build.opensuse.org/package/live_build_log/games/urbanterror/openSUSE_Tumbleweed/aarch64
> code/server/../qcommon/q_platform.h:272:2: error: #error "Architecture not supported"

File: urbanterror.appdata.xml (hunk 4 above)
Comment (Mailaender) [reply urbanterror.appdata.xml:5]: Engine is GPL, game scripts and assets are under this weird Quake III SDK license which only allows digital distribution for free.

File: info.urbanterror.UrbanTerror.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+rename-icon: urbanterror
+rename-desktop-file: urbanterror.desktop
+rename-appdata-file: urbanterror.appdata.xml
+finish-args:
+  - --socket=pulseaudio
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --filesystem=~/.q3a/q3ut4/
```
Comment (barthalion) [inline info.urbanterror.UrbanTerror.yaml:15]: Any reason this is `--filesystem` and not `--persist=.q3a`?

File: info.urbanterror.UrbanTerror.yaml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+modules:
+  - shared-modules/glu/glu-9.json
+  - shared-modules/SDL/SDL-1.2.15.json
+
+  - name: ioq3-urbanterror
+    only-arches:
+      - x86_64
```
Comment (barthalion) [inline info.urbanterror.UrbanTerror.yaml:23]: Nitpick: this is not needed with flathub.json

File: info.urbanterror.UrbanTerror.yaml (hunk 6 above)
Comment (Mailaender) [reply info.urbanterror.UrbanTerror.yaml:15]: Thanks. That fixes the game not saving configuration (asking for a player name on every start)

**Comment by Mailaender** [comment]
bot, build info.urbanterror.UrbanTerror

**Comment by pizzadude** [comment]
https://github.com/flathub/info.urbanterror.UrbanTerror/issues/1

**Comment by Mailaender** [comment]
Stop cross-posting and pinging me all the time. I am subscribed to all of that.

**Pull request closed without merging**